src/consensus/blockstore.rs

```diff
@@ -265,70 +138,105 @@ impl Blockstore {
     /// Returns `None` if blockstore does not hold any version of the block yet.
     #[must_use]
     pub fn canonical_block_hash(&self, slot: Slot) -> Option<Hash> {
-        self.canonical.get(&slot).copied()
+        self.slot_data(slot)?
+            .canonical
+            .completed
+            .clone()
+            .map(|c| c.0)
```

Comment:
```suggestion
        self.slot_data(slot)?
            .canonical
            .completed
            .map(|c| c.0.clone())
```

so we do not end up cloning the `Block` unnecessarily.